src/components/side-nav/SideNav.jsx

```diff
@@ -1,5 +1,6 @@
 import React from "react";
 
+import { observer } from "mobx-react-lite";
```

Comment:
"observer", wraps the component's render function in mobx, autorun to make sure that any data that is used during the rendering of a component forces a re-rendering upon change.